client/src/resources/asset.rs

```diff
@@ -253,6 +255,97 @@ mod asset_identifier_tests {
     }
 }
 
+/// When a bad token or string is provided to parsing into an asset
+/// you get an error.
+#[derive(Debug)]
+pub struct ParseAssetError {
+    details: String,
```

Comment:
Strings make for bad error typing. You should use an Enum with the possible errors coded into it. Especially as there is only one error possible.